src/main/java/org/eclipse/yasson/model/PropertyModel.java

```diff
@@ -133,24 +133,46 @@ private AdapterBinding getUserAdapterBinding(Property property, JsonbContext jso
 
     private PropertyCustomization introspectCustomization(Property property, JsonbContext jsonbContext) {
         final AnnotationIntrospector introspector = jsonbContext.getAnnotationIntrospector();
-        final CustomizationBuilder builder = new CustomizationBuilder();
+        final PropertyCustomizationBuilder builder = new PropertyCustomizationBuilder();
         //drop all other annotations for transient properties
         if (introspector.isTransient(property)) {
             builder.setJsonbTransient(true);
             return builder.buildPropertyCustomization();
         }
         builder.setJsonReadName(introspector.getJsonbPropertyJsonReadName(property));
         builder.setJsonWriteName(introspector.getJsonbPropertyJsonWriteName(property));
-        builder.setNillable(classModel.getClassCustomization().isNillable()
-                || introspector.isPropertyNillable(property));
+        builder.setNillable(classModel.getClassCustomization().isNillable() || introspector.isPropertyNillable(property));
         builder.setAdapterInfo(getUserAdapterBinding(property, jsonbContext));
         builder.setSerializerBinding(getUserSerializerBinding(property, jsonbContext));
         builder.setDeserializerBinding(introspector.getDeserializerBinding(property));
         builder.setDateFormatter(introspector.getJsonbDateFormat(property));
-        builder.setNumberFormat(introspector.getJsonbNumberFormat(property));
+        introspectNumberFormatter(property, introspector, builder);
         return builder.buildPropertyCustomization();
     }
 
+    private void introspectNumberFormatter(Property property, AnnotationIntrospector introspector, PropertyCustomizationBuilder builder) {
+        /*
+         * If @JsonbNumberFormat is placed on getter implementation must use this format on serialization.
+         * If @JsonbNumberFormat is placed on setter implementation must use this format on deserialization.
+         * If @JsonbNumberFormat is placed on field implementation must use this format on serialization and deserialization.
+         */
+        Map<AnnotationTarget, JsonbNumberFormatter> jsonNumberFormatCategorized = introspector.getJsonNumberFormatter(property);
+        if(jsonNumberFormatCategorized.keySet().contains(AnnotationTarget.PROPERTY)) {
```

Comment:
Yes, this applies to all customization overrides according to spec, specifically for number format, there is a statement:

Annotation applied to more specific target overrides the same annotation applied to target with wider scope. This would mean "getter / settrer > field > class > package > global configuration" as "most specific > least specific".